### PR TITLE
Add configureDeploymentSource to Deployments Node

### PR DESCRIPTION
**Pull request by nturinski**
Fixes https://github.com/Microsoft/vscode-azureappservice/issues/727

Adds telemetry so we know what's the most common user entry point.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -206,11 +207,9 @@ export async function activateInternal(
         registerCommand('appService.Deploy', async function (this: IActionContext, target?: vscode.Uri | WebAppTreeItem | undefined): Promise<void> {
             await deploy(this, true, target);
         });
-        registerCommand('appService.ConfigureDeploymentSource', async function (this: IActionContext, node?: SiteTreeItem): Promise<void> {
-            if (!node) {
-                node = <SiteTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
-            }
-            await editScmType(node.root.client, node, this);
+        registerCommand('appService.ConfigureDeploymentSource', async function (this: IActionContext): Promise<void> {
+            await editScmType(this);
```
Comment (EricJizbaMSFT) [inline src/extension.ts]: Shouldn't there be a `node` to pass along?

File: src/commands/deployments/editScmType.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export async function editScmType(actionContext: IActionContext, node?: SiteTreeItem | appservice.DeploymentsTreeItem): Promise<void> {
+    if (!node) {
+        node = <SiteTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+    } else if (node instanceof appservice.DeploymentsTreeItem) {
+        node = <SiteTreeItem>node.parent;
+    }
+
+    await appservice.editScmType(node.root.client, node, actionContext);
+    const children: AzureTreeItem[] = await node.getCachedChildren();
+    const deploymentsTreeItem: AzureTreeItem | undefined = children.find(ti => {
```
Comment (EricJizbaMSFT) [inline src/commands/deployments/editScmType.ts]: You can just use this to get the deployments node:
https://github.com/Microsoft/vscode-azureappservice/blob/master/src/explorer/SiteTreeItem.ts#L27

File: src/commands/deployments/editScmType.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import * as appservice from "vscode-azureappservice";
```
Comment (EricJizbaMSFT) [inline src/commands/deployments/editScmType.ts]: nit: copyright header

File: src/commands/deployments/editScmType.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import { AzureTreeItem, IActionContext } from "vscode-azureextensionui";
+import { SiteTreeItem } from "../../explorer/SiteTreeItem";
+import { WebAppTreeItem } from "../../explorer/WebAppTreeItem";
+import { ext } from "../../extensionVariables";
+
+export async function editScmType(actionContext: IActionContext, node?: SiteTreeItem | appservice.DeploymentsTreeItem): Promise<void> {
+    if (!node) {
+        node = <SiteTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+    } else if (node instanceof appservice.DeploymentsTreeItem) {
```
Comment (EricJizbaMSFT) [inline src/commands/deployments/editScmType.ts]: It looks like you removed the context menu from SiteTreeItem. Do you need to check this anymore?

Comment (nturinski) [reply src/commands/deployments/editScmType.ts]: Yeah because the node picker will still return a SiteTreeItem

Comment (EricJizbaMSFT) [reply src/commands/deployments/editScmType.ts]: Then change the node picker to get the deployments tree item?

**Comment by nturinski** [comment]
@fiveisprime Are we fine to remove this from the web app context menu and have it only on the Deployments context menu?

File: src/commands/deployments/editScmType.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import * as appservice from "vscode-azureappservice";
+import { IActionContext } from "vscode-azureextensionui";
+import { SiteTreeItem } from "../../explorer/SiteTreeItem";
+import { ext } from "../../extensionVariables";
+
+export async function editScmType(actionContext: IActionContext, node?: SiteTreeItem | DeploymentsTreeItem): Promise<void> {
+    if (!node) {
+        node = <DeploymentsTreeItem>await ext.tree.showTreeItemPicker([DeploymentsTreeItem.contextValueConnected, DeploymentsTreeItem.contextValueUnconnected]);
+    } else if (node instanceof SiteTreeItem) {
+        actionContext.properties.contextValue = node.contextValue;
```
Comment (EricJizbaMSFT) [inline src/commands/deployments/editScmType.ts]: No need to do this. It's handled here: https://github.com/Microsoft/vscode-azuretools/blob/de0a8e1a1c17ff9f5b3b4fecc53b163cb64e3625/ui/src/AzureActionHandler.ts#L30

Comment (nturinski) [reply src/commands/deployments/editScmType.ts]: Your knowledge of the codebase is impressive as always 😅

**Pull request merged**